includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -79,7 +81,9 @@ public function set_api_credentials( $credentials ) {
 			$ping = null;
 		}
 		return $ping ?
-			update_option( 'newspack_newsletters_mailchimp_api_key', $api_key ) :
+			// 'newspack_mailchimp_api_key' is a new option introduced to manage MC API key accross Newspack plugins.
+			// Keeping the old option for backwards compatibility.
```

Comment:
Nit: this comment is irrelevant in this case, there's no legacy option name handling here.